controllers/openstackdataplanenode_controller.go

```diff
@@ -453,3 +453,9 @@ func (r *OpenStackDataPlaneNodeReconciler) GetAnsibleSSHPrivateKeySecret(instanc
 	}
 	return instanceRole.Spec.NodeTemplate.AnsibleSSHPrivateKeySecret
 }
+
+// GetServices returns the list of services for the node's role
+// Note that these are not inherited from NodeTemplate.
+func (r *OpenStackDataPlaneNodeReconciler) GetServices(instance *dataplanev1beta1.OpenStackDataPlaneNode, instanceRole *dataplanev1beta1.OpenStackDataPlaneRole) []string {
+	return instanceRole.Spec.Services
+}
```

Comment:
shouldn't we get `instance.Spec.Services` when the node is deployed individually?

Reply:
i've just set the services on the role for now. I did not make this an inherited field in NodeTemplate initially. but perhaps we want to?